### DISCUSSION TITLE
refactor: noasm to purego

### DISCUSSION THREAD

**Discussion by leongross** [klauspost/compress#990] [Ideas/Features]
Is there a good reason to use the `noasm` tag in this project contrary to adapting the build tag `purego`? As I see it, they both mean the same thing or am I confusing concepts here?
The Golang Standard Library utilizes this build tags for e.g. [generic cryptography implementations](https://cs.opensource.google/go/go/+/master:src/syscall/syscall.go;l=45;drc=94982a07825aec711f11c97283e99e467838d616).
It might be a good idea to adjust this project's build tags to the widely adopted scheme.

**Comment by klauspost** [comment]
First of all we can't change it. Second, as you point out it is pointless to have both.

**Reply by leongross** [reply]
What do you mean we can't change it? Why can't the `noasm` tag be adjusted to `purego`? What is the blocker? I see that other projects that include compress as a dependency need to adjust this, but adjusting to purego and reducing the set of build tags developers have to take care of while providing the same functionality sounds like a good idea.

**Reply by klauspost** [reply]
Because people are using it.

**Reply by leongross** [reply]
> as you point out it is pointless to have both

I don't think it is useless. Aliasing `purego` to `noasm` (`go:build (noasm || purego)...`) guarantees backward compatibility and also adapts to newer conventions. I only see benefits for developers, since they do not have to look up how library x enables pure go support but can just go the default way with the `purego` tag.
